### ISSUE TITLE
SAM output quality values degenerate

### ISSUE THREAD

**Issue by map2085** [PacificBiosciences/blasr#48]
I give a .fastq to BLASR and set the flag "-printSAMQV", and would like my SAM output to have the appropriate base quality scores.

However, in the SAM output file from BLASR, the Quality field is empty (i.e. just an asterisk "*"), and instead there seems to be a quality "tag" ( "ts:Z:<qualities>"  ).

But for every read/alignment, this quality tag consists only of a string of "5" of the same length as the read  (e.g.  "ts:Z:555555......").  Please note that these are not the qualities that appeared in the .fastq input.

In other words, BLASR seems to lose the quality information and instead output degenerate qualities.

Thank you.

**Comment by ylipacbio** [comment]
Blasr does not propagate Quality values from FASTQ files to alignments.

In order to get alignments with PacBio QVs, first of all, you will need to **use bax/bas.h5 as input**. Then you can choose one of the following ways to generate alignments with QVs.

(1) Call pbalign with --loadQVs (or alias --forQuiver) to generate output cmp.h5 file. All PacBio QV info will be contained in the output.cmp.h5 file, which can be opened via SMRTView or  pbcore.io.CmpH5Reader. e.g., 

    pbalign input_bax.h5 reference.fasta output.cmp.h5 --loadQVs

(2) Call blasr with -sam -printSAMQV to generate output sam file. In this case, you will have to **use the refactored blasr code base** at

    https://github.com/PacificBiosciences/refactored_blasr

since we are deprecating this blasr code base now.
